src/app/shell/result/result.component.ts

```diff
@@ -63,14 +70,30 @@ export class ResultComponent implements OnInit, OnDestroy {
     private navigationBarService: NavigationBarService,
     private route: ActivatedRoute,
     private router: Router
-  ) {}
+  ) {
+    this.route.queryParamMap
+      .pipe(
+        tap((params: ParamMap) => {
+          const filterParams = params.get('filter');
+          if (filterParams) {
+            this.store.dispatch(new SetFilterFromURL(Util.parseFilterStateQuery(filterParams)));
+          }
+        }),
+        takeUntil(this.destroy$)
+      )
+      .subscribe();
```

Comment:
why do you prefer to do this functionality in tap instead of subscribe? do we need this action one more time before ngOndestroy (if I get this approach correctly)?

Reply:
let us move it to subscribe. Tap is redundant